core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -107,6 +124,14 @@ public ByteBuffer allocate(int size) {
         return bb;
     }
 
+  // Allocates Slice, meaning it must be known from which block it is allocated.
+  // Because currently the free list doesn't keeps block IDs for released values,
```

Comment:
Because in the future the memory management is planned to be via slices only, so this is kind of preparation. I agree this is not necessary now, but I prefer to keep it this way.